__tests__/main.test.ts

```diff
@@ -0,0 +1,379 @@
+    });
+
+    test('PipelineHelper.equals() - return if strings are equal', () => {
+        expect(PipelineHelper.equals(null, null)).toBeTruthy();
+        expect(PipelineHelper.equals('a', null)).toBeFalsy();
+        expect(PipelineHelper.equals(null, 'a')).toBeFalsy();
+        expect(PipelineHelper.equals('a', 'a ')).toBeTruthy();
+        expect(PipelineHelper.equals('a', 'A')).toBeTruthy();
+        expect(PipelineHelper.equals('a', 'b')).toBeFalsy();
+    });
+
+    test('PipelineHelper.processEnv() - return specified env variable varaiable', () => {
+        process.env['envVar'] = 'value';
+        expect(PipelineHelper.processEnv('envVar')).toBe('value');
+    });
+
+    test('PipelineHelper.processEnv() - throw error if specified envVar is not available', () => {
+        process.env['envVar'] = '';
+        expect(() => PipelineHelper.processEnv('envVar')).toThrow(`env.${'envVar'} is not set`);
+    });
+
+    test('PipelineHelper.isGitHubArtifact() - returns if artifact if of type github', () => {
+        expect(PipelineHelper.isGitHubArtifact({})).toBeFalsy();
+        expect(PipelineHelper.isGitHubArtifact({ type: 'githuB' })).toBeTruthy();
+        expect(PipelineHelper.isGitHubArtifact({ type: null })).toBeFalsy();
+    });
+
+    test('PipelineHelper.getErrorAndWarningMessageFromBuildResult() - concatenate and return errors', () => {
+        expect(PipelineHelper.getErrorAndWarningMessageFromBuildResult([
+            { message: 'FirstMessage', result: 2},
+            { message: 'FirstIgnoredMessage', result: 0},
+            { message: 'SecondIgnoredMessage', result: 1},
+            { message: 'SecondMessage', result: 2},
+        ])).toMatchObject({
+            errorMessage: 'FirstMessage,SecondMessage',
+            warningMessage: '',
+        });
+    });
+
+    test('PipelineHelper.getErrorAndWarningMessageFromBuildResult() - concatenate and return warnings if no errors', () => {
+        expect(PipelineHelper.getErrorAndWarningMessageFromBuildResult([
+            { message: 'FirstIgnoredMessage', result: 0},
+            { message: 'SecondIgnoredMessage', result: 1},
+        ])).toMatchObject({
+            errorMessage: '',
+            warningMessage: 'FirstIgnoredMessage,SecondIgnoredMessage',
+        });
+    });
+
+    test('PipelineHelper.getErrorAndWarningMessageFromBuildResult() - server errors also which comes not in form of array', () => {
+        expect(PipelineHelper.getErrorAndWarningMessageFromBuildResult(
+            { message: 'ErrorMessage' } as any
+        )).toMatchObject({
+            errorMessage: 'ErrorMessage',
+            warningMessage: '',
+        });
+    });
+
+    test('PipelineHelper.getErrorAndWarningMessageFromBuildResult() - server errors also which comes not in form of array', () => {
```

Comment:
what is being tested here?

Reply:
As in? I was trying to capture the different if clauses to show how the `_getErrorMessageFromServer` functino parses the result object to return error messages.

Reply:
2 tests have the same description - specify exactly what is tested in each in short